server/utils/textTracks.ts

```diff
@@ -0,0 +1,187 @@
+    return {
+      track,
+      prevId: existingTrack.id,
+    };
+  });
+
+  /**
+   * For tracks that were updated but didn't include a new user-sent file,
+   * we can't point to a stored track in Strapi. Instead, we need to point to the
+   * track hosted in Mux itself.
+   **/
+  const alreadyInMuxWithoutFile = await Promise.all(
+    updatedTracks.flatMap((t) => {
+      const track_id = t.track.stored_track?.id;
+      if (t.track.file || !track_id) return [];
+
+      return new Promise<(typeof updatedTracks)[number]>(async (resolve, reject) => {
+        try {
+          const muxTrackRes = await fetch(
+            getMuxTextTrackUrl({ playback_id, track: { id: track_id }, signedToken: token })
+          );
+
+          if (muxTrackRes.status !== 200) throw new Error(muxTrackRes.statusText);
+
+          const contents = await muxTrackRes.text();
+          const type = muxTrackRes.headers.get('content-type') || 'text/vtt';
+
+          const contentLength = muxTrackRes.headers.get('content-length');
+
+          const file = TextTrackFile.parse({
+            contents,
+            type,
+            name: t.track.name,
+            size: contentLength && !Number.isNaN(Number(contentLength)) ? Number(contentLength) : contents.length,
+          } as z.input<typeof TextTrackFile>);
+
+          resolve({
+            ...t,
+            track: {
+              ...t.track,
+              file,
+            },
+          });
+        } catch (error) {
+          reject(error);
+        }
+      });
+    })
+  );
+
+  const newStoredForDownload = await (async () => {
+    try {
+      return await storeTextTracks(
+        [...addedTracks, ...updatedTracks.map((t) => t.track), ...alreadyInMuxWithoutFile.map((t) => t.track)].filter(
+          (t) => t.file
+        ) as ParsedCustomTextTrack[]
+      );
+    } catch (error) {
+      console.error('\n\n[updateTextTracks / storeTextTracks]', error);
+      throw new Error('Unable to store text tracks');
+    }
+  })();
+
+  await (async () => {
+    try {
+      await getService('mux').deleteAssetTextTracks(asset_id, [
+        ...removedTracks.map((t) => t.id),
+        ...updatedTracks.map((t) => t.prevId),
+      ]);
+    } catch (error) {
+      console.error('\n\n[updateTextTracks / deleteAssetTextTracks]', error);
+      throw new Error('Unable to delete text tracks in Mux');
+    }
+  })();
+
+  await (async () => {
+    try {
+      await getService('mux').createAssetTextTracks(asset_id, newStoredForDownload.map(storedTextTrackToMuxTrack));
+    } catch (error) {
+      console.error('\n\n[updateTextTracks / createAssetTextTracks]', error);
+      throw new Error('Unable to create text tracks in Mux');
+    }
+  })();
+
+  // Give mux' servers enough time to download tracks before deleting them
+  if (alreadyInMuxWithoutFile.length > 0) {
+    await sleep(1000);
+  }
+}
+
+export function sleep(ms: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
```

Comment:
This feels like it should be utilizing our webhooks to determine if the file was downloaded (or failed):

Webhook events I would use to determine this:
- https://docs.mux.com/webhook-reference#video.asset.track.ready
- https://docs.mux.com/webhook-reference#video.asset.track.errored

Reply:
This wasn't being used, actually - it was a leftover from an initial iteration